components/org.wso2.carbon.identity.governance/src/main/java/org/wso2/carbon/identity/governance/IdentityGovernanceServiceImpl.java

```diff
@@ -88,10 +99,12 @@ public void updateConfiguration(String tenantDomain, Map<String, String> configu
             residentIdp.setFederatedAuthenticatorConfigs(configsToSave.toArray(new
                     FederatedAuthenticatorConfig[configsToSave.size()]));
             identityProviderManager.updateResidentIdP(residentIdp, tenantDomain);
+        } catch (IdentityProviderManagementClientException e) {
+            log.debug("Client error while updating identityManagement Properties of Resident Idp.", e);
+            throw new IdentityGovernanceClientException(e.getMessage(), e);
```

Comment:
Do we send a proper message in the exception itself?

Reply:
Yes
example : 
<img width="1039" alt="Screenshot 2024-06-19 at 18 08 28" src="https://github.com/wso2-extensions/identity-governance/assets/42939752/bb364b9c-7576-4b88-b297-123bda74b733">